File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.py
class Solution:
    def reverseWords(self, s: str) -> str:
        # Split the string
        s = s.split()
        # Convert string into a list
        words = list(s)
        left, right = 0, len(words) - 1
        while left < right:
            # Swap words
            s[left], s[right] = s[right], s[left]
            left += 1
            right -= 1
        # Convert list back to string
        return " ".join(s)
        

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {
        // Trim the input string to remove leading and trailing spaces
        String[] words = s.trim().split("\\s+");
        int left = 0;
        int right = words.length - 1;
        while (left < right) {
            String temp = words[left];
            words[left] = words[right];
            words[right] = temp;
            left++;
            right--;
        }
        
        // Join the reversed words array into a single string using a space as the separator
        return String.join(" ", words);
    }

}